Spanner HTTP emulator fails to start. "Address already in use"

* GCB: https://console.cloud.google.com/cloud-build/builds;region=us-east1/25085c2f-0f35-4bd1-92e8-0b5180cc1b6c;tab=detail?project=cloud-cpp-testing-resources
* Raw: https://storage.googleapis.com/cloud-cpp-community-publiclogs/logs/google-cloud-cpp/main/d27e4ffda2e291b5a681da2e4787be7c9bcbbda1/fedora-37-cxx14-cxx14/log-25085c2f-0f35-4bd1-92e8-0b5180cc1b6c.txt

```
2023-03-21T05:28:25Z (+213s): Spanner emulator started at localhost:40123
2023-03-21T05:28:33Z (+221s): Cannot determine Cloud Spanner HTTP emulator port.
2023/03/21 05:28:25 gateway.go:115: Error waiting for emulator to start: emulator failed to come up at localhost:40125 within 30s deadline: rpc error: code = Unimplemented desc = Method not found: google.spanner.admin.instance.v1.InstanceAdmin/ListInstanceConfigs
E0321 05:28:25.357047068   39558 server_chttp2.cc:48]        {"created":"@1679376505.357009985","description":"No address added out of total 2 resolved","file":"external/com_github_grpc_grpc/src/core/ext/transport/chttp2/server/chttp2_server.cc","file_line":873,"referenced_errors":[{"created":"@1679376505.356994013","description":"Unable to configure socket","fd":5,"file":"external/com_github_grpc_grpc/src/core/lib/iomgr/tcp_server_utils_posix_common.cc","file_line":217,"referenced_errors":[{"created":"@1679376505.356989523","description":"Address already in use","errno":98,"file":"external/com_github_grpc_grpc/src/core/lib/iomgr/tcp_server_utils_posix_common.cc","file_line":190,"os_error":"Address already in use","syscall":"bind"}]},{"created":"@1679376505.357007199","description":"Address family not supported by protocol","errno":97,"file":"external/com_github_grpc_grpc/src/core/lib/iomgr/socket_utils_common_posix.cc","file_line":446,"os_error":"Address family not supported by protocol","syscall":"socket","target_address":"[::1]:40125"}]}
2023-03-21 05:28:25  ERROR  server.cc : 318 : Failed to bind to address: localhost:40125
2023-03-21 05:28:25  ERROR  emulator_main.cc : 35 : Failed to start gRPC server.
```

Error says "address already in use". I guess something was already running on port :40125 ?

Looks like we start one emulator on `localhost:0` https://github.com/googleapis/google-cloud-cpp/blob/d27e4ffda2e291b5a681da2e4787be7c9bcbbda1/google/cloud/spanner/ci/lib/spanner_emulator.sh#L92

Then it gets assigned a port? and we use the next two ports for the http emulator:  https://github.com/googleapis/google-cloud-cpp/blob/d27e4ffda2e291b5a681da2e4787be7c9bcbbda1/google/cloud/spanner/ci/lib/spanner_emulator.sh#L117-L121

--- a/google/cloud/spanner/ci/lib/spanner_emulator.sh
+++ b/google/cloud/spanner/ci/lib/spanner_emulator.sh
@@ -87,7 +87,7 @@ function spanner_emulator::start() {
   fi
 
   # The tests typically run in a Docker container, where the ports are largely
-  # free; when using in manual tests, you can set EMULATOR_PORT.
+  # free; when using in manual tests, you can set SPANNER_EMULATOR_HOST.
   rm -f emulator.log
   "${SPANNER_EMULATOR_CMD}" --host_port "localhost:${emulator_port}" >emulator.log 2>&1 </dev/null &
   SPANNER_EMULATOR_PID=$!
@@ -111,15 +111,21 @@ function spanner_emulator::start() {
     return 1
   fi
 
-  # The tests typically run in a Docker container, where the ports are largely
-  # free; when using in manual tests, you can set REST_EMULATOR_PORT.
-  rm -f http_emulator.log
-  # Leverage the previously determined emulator_port to avoid collision.
   local http_emulator_port=$((emulator_port + 1))
+  if [[ $# -ge 2 ]]; then
+    http_emulator_port=$2
+  fi
   local grpc_emulator_port=$((http_emulator_port + 1))
+  if [[ $# -ge 3 ]]; then
+    grpc_emulator_port=$3
+  fi
+
+  # The tests typically run in a Docker container, where the ports are largely
+  # free; when using in manual tests, you can set SPANNER_EMULATOR_REST_HOST.
+  rm -f http_emulator.log
   "${SPANNER_HTTP_EMULATOR_CMD}" --hostname "localhost" \
     --grpc_port "${grpc_emulator_port}" --http_port "${http_emulator_port}" \
-    >http_emulator.log 2>&1 </dev/null &
+    --copy_emulator_stdout=true >http_emulator.log 2>&1 </dev/null &
   SPANNER_HTTP_EMULATOR_PID=$!
 
   http_emulator_port="$(spanner_emulator::internal::read_http_emulator_port http_emulator.log)"

--- a/google/cloud/spanner/ci/run_integration_tests_emulator_cmake.sh
+++ b/google/cloud/spanner/ci/run_integration_tests_emulator_cmake.sh
@@ -38,7 +38,7 @@ ctest_args=("$@")
 cd "${CMAKE_BINARY_DIR}"
 
 # Starts the emulator and arranges to kill it.
-spanner_emulator::start
+spanner_emulator::start 8787
 trap spanner_emulator::kill EXIT
 
 ctest -R "^spanner_" "${ctest_args[@]}"

